src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let join = '';
+
+    for (const item of this) {
+      if (item !== undefined && item !== null) {
+        join += item;
+      }
+
+      if (this.indexOf(item) !== this.length - 1) {
+        join += separator;
+      }
```

Comment:
suggestion: I'd rather use for-with-indexes loop to avoid using `indexOf`, but still – up to you